allennlp/training/trainer.py

```diff
@@ -411,6 +411,10 @@ def train(self) -> Dict[str, Any]:
         epochs_trained = 0
         training_start_time = time.time()
 
+        metrics['best_epoch'] = self._metric_tracker.best_epoch(epoch_counter - 1)
+        for key, value in self._metric_tracker.best_epoch_metrics.items():
+            metrics["best_validation_" + key] = value
+
```

Comment:
shouldn't this logic live inside `_restore_checkpoint()`?

Reply:
yeah, that would be the right place. Shifted it. 👍

Reply:
oh, I see, yeah, I was wrong about this :frowning_face:

Reply:
I think I was thinking that `metrics` was an instance variable, so you could have just updated it in _restore_checkpoint, but it's not

Reply:
Ohh K, right. Cool, no problem! 👍